src/main/java/com/notably/logic/parser/NewCommandParser.java

```diff
@@ -52,7 +52,12 @@ public NewCommandParser(Model notablyModel) {
         }
 
         AbsolutePath path = ParserUtil.createAbsolutePath(title, notablyModel.getCurrentlyOpenPath());
-        Block block = new BlockImpl(new Title(title), new Body(body));
+        Block block;
+        try {
+            block = new BlockImpl(new Title(title), new Body(body));
+        } catch (IllegalArgumentException ex) {
+            throw new ParseException(ex.getMessage());
+        }
 
         List<Command> commands = new ArrayList<>();
```

Comment:
You want to do a defensive check that block is not null?

Reply:
good point, i've added a check for null . Thank you !